model-optimizer/extensions/back/names_uniqueness_check.py

```diff
@@ -1,40 +1,41 @@
 # Copyright (C) 2018-2021 Intel Corporation
 # SPDX-License-Identifier: Apache-2.0
-
+from collections import defaultdict
 from extensions.back.pass_separator import BackFinish
 from mo.back.replacement import BackReplacementPattern
-from mo.graph.graph import Graph, rename_node, Node
+from mo.graph.graph import Graph, rename_node
 
 
-def update_dictionaries(names_dict: dict, not_unique_names: set, node: Node, node_name: str):
-    if node_name not in names_dict.keys():
-        names_dict[node_name] = [node]
-    else:
+def nodes_with_equal_names(graph: Graph):
+    names_dict = defaultdict(list)
+    for node in graph.get_op_nodes():
+        node_name = node.soft_get('name', node.id)
         names_dict[node_name].append(node)
-        not_unique_names.add(node_name)
+    return names_dict
 
 
-def rename_not_unique_nodes(names_dict_1: dict, names_dict_2: dict, not_unique_names: set, skip_result: bool = True):
-    for name in not_unique_names:
-        for idx, node in enumerate(names_dict_1[name]):
-            new_node_name = name + '_{}'.format(idx)
+def make_node_names_unique(nodes: list, node_names: set):
+    results_pos = [idx for idx, node in enumerate(nodes) if node.op == 'Result']
```

Comment:
Please, add comments to the functions describing their specifics. Especially for the Result node. I have provided some details in the comment.